paddle/fluid/platform/mkldnn_reuse.h

```diff
@@ -676,12 +678,12 @@ class TransposeMKLDNNHandler : public MKLDNNHandler {
       auto dst_mdp = mkldnn::memory::primitive_desc{
           Axis2MemoryDesc(dims_, axis_), engine_};
 
-      auto dst_data = output->mutable_data<float>(place, dst_mdp.get_size());
+      auto dst_data = output->mutable_data<T>(place);
```

Comment:
Please restore the removed `dst_mdp.get_size()`. Not setting the size recommended by MKL-DNN but having Paddle compute it, leads to errors such as in #20998.